expense-splitter/src/components/expense/ExpenseDetail.jsx

```diff
@@ -1,27 +1,135 @@
+import { useEffect, useRef, useState } from "react";
 import { useParams, useNavigate } from "react-router-dom";
 import { UseDataContext } from "../context/SiteContext";
-import Button from "../ui/Button";
+import Card from "../ui/CardExpenseDetail";
 
 function ExpenseDetail() {
-  const { expenses } = UseDataContext();
+  const { expenses, groupData, friends } = UseDataContext();
   const { expenseId } = useParams();
   const navigate = useNavigate();
 
-  const singleExpense = expenses.find((expense) => expense.id === expenseId);
+  // get expense details
+  const expenseDetails = expenses.find((expense) => expense.id === expenseId);
+
+  // get total amount of expense
+  const totalAmount = expenseDetails.amount;
+
+  // get group connected to expense
+  const expenseGroup = groupData.filter(
+    (group) => group.id === expenseDetails.groupId,
+  );
+
+  // turn array into single object
+  const findGroup = expenseGroup.find(
+    (group) => group.id === expenseDetails.groupId,
+  );
+
+  // get friends names from goup
+  const getFriends = friends.filter((friend) =>
+    findGroup.friendIDs.includes(friend.id),
+  );
+
+  // get weight array from expense object
+  const getWeight = expenseDetails.weight;
+
+  // set data for pie chart to be array of contribution values
+  const pieChartData = getWeight.map((item) =>
+    ((item.percentage / 100) * totalAmount).toFixed(2),
+  );
+
+  // sort payers by contribution amount
+  const sortedContributions = getWeight.sort(function (a, b) {
+    return b.percentage - a.percentage;
+  });
+
+  // create PieChart function
+  function PieChart() {
```

Comment:
There is a PieChart component in the widgets folder. I would recommend trying to use that to reduce code duplication

Reply:
the pie charts are very similar, but the data and labels are different.

Do I create a props that get passed to ExpenseDetail that resembles the data/label inputs?

Reply:
Make sure you have the latest PieChart component from dev

Then pass in the required props for the pie chart. It looks like it takes 2 props, a label and a the data

The label is a string of what the slices represent and the data would look something like this:
{
friend1: amount owed,
friend2: amount owed
}

I believe Carlos implemented it. You could check with him if it's giving you trouble

Reply:
after we merge this to dev I can clean this up